frontend/src/components/NewVocabForm.tsx

```diff
@@ -34,8 +71,12 @@ const NewVocabFormComponent = () => {
                     <button type="submit">Submit</button>
                 </form>
                 <div>
-                    {responseText.split('\n').map((line, index) => (
-                        <p key={index}>{line}</p>
+                    {sentences.map((item, index) => (
+                        <div key={index}>
+                            <p>{item.word}: {item.sentence}</p>
```

Comment:
each item should now also has a sentence_id, which should be passed into handleSave and handleFavorite